integration-tests/scaffolding/helpers.ts

```diff
@@ -1,22 +1,166 @@
+
+interface BeginEndBalances {
+    beginBalance: bigint;
+    endBalance?: bigint;
+}
+let balancesMap = new Map<string, Map<string, BeginEndBalances>>;
+let doTxAccounting = false;
+export let txAccountingHook = async (api, context?): Promise<void> => { };
+
+if (process.env.ENABLE_TX_ACCOUNTING === 'true' || process.env.ENABLE_TX_ACCOUNTING === '1') {
+    doTxAccounting = true;
+    txAccountingHook = async (api, context?) => showTotalCost(api, context);
+}
+
+export function parseResult() {
     return pipe(
-        map((result: ISubmittableResult) => result.events.reduce((acc, {event}) => { acc[eventKey(event)] = event.data; return acc}, {})),
-    )
+        tap((result: ISubmittableResult) => {
+            if (result.dispatchError) {
+                let err: EventError;
+
+                switch (result.dispatchError.type) {
+                    case 'Module': {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+
+                    default: {
+                        err = new EventError(result.dispatchError);
+                        break;
+                    }
+                }
+
+                console.log(err.toString());
+                throw err;
+            }
+        }),
+        map((result: ISubmittableResult) => result.events.reduce((acc, { event }) => { acc[eventKey(event)] = event.data; return acc }, {})),
+        // tap((events) => console.log(events)),
+    );
 }
 
-export function eventKey(event:Event): string {
+export function eventKey(event: Event): string {
     return `${event.section}.${event.method}`;
 }
 
 export function signPayloadSr25519(keys: KeyringPair, data: Codec): Sr25519Signature {
-    return {Sr25519: u8aToHex(keys.sign(u8aWrapBytes(data.toU8a())))}
-}
\ No newline at end of file
+    return { Sr25519: u8aToHex(keys.sign(u8aWrapBytes(data.toU8a()))) }
+}
+
+export function createAccount(name: string = 'first pair') {
+    const mnemonic = mnemonicGenerate();
+    // create & add the pair to the keyring with the type and some additional
+    // metadata specified
+    const keyring = new Keyring({ type: 'sr25519' });
+    const keypair = keyring.addFromUri(mnemonic, { name }, 'sr25519');
+
+    return keypair;
+}
+
+export async function createAndFundAccount({ api, amount, source, name, context }: AccountFundingInputs) {
+    const keypair = createAccount(name);
+
+    // Get keypair for pre-funded dev account
+    const devKeys = createKeys(source);
+
+    // Transfer funds from pre-funded dev account to new account
+    await transferFunds(api, devKeys, keypair, amount);
+
+    if (doTxAccounting) {
+        let map = balancesMap.get(context || 'undefined');
+        if (!map) {
+            map = new Map<string, BeginEndBalances>();
+        }
+
+        map.set(keypair.address, { beginBalance: amount });
+        balancesMap.set(context || 'undefined', map);
```

Comment:
Are we setting the string 'undefined' to a map key? Is this to get a default?

Reply:
Yes. In cases where a context was not set, we'll display it as 'undefined'.